src/App.jsx

```diff
@@ -14,33 +15,87 @@
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_BY_ALPHABETICAL = 'alphabetical';
+const SORT_BY_LENGHT = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  function sortGoods(goods, sortField, reverse) {
+    const changedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    if (sortField) {
+      changedGoods.sort((good1, good2) => {
+        switch (sortField) {
+          case SORT_BY_ALPHABETICAL:
+            return good1.localeCompare(good2);
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+          case SORT_BY_LENGHT:
+            return good1.length - good2.length;
+
+          default:
+            return 0;
+        }
+      });
+    }
+
+    if (reverse) {
+      return changedGoods.reverse();
+    }
+
+    return changedGoods;
+  }
+
+  const [sortMethod, setSortMethod] = useState('');
+  const [isReverse, setIsReverse] = useState(false);
+  const visibleGoods = sortGoods(goodsFromServer, sortMethod, isReverse);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button ${sortMethod === SORT_BY_ALPHABETICAL ? 'button is-info' : 'button is-info is-light'}`}
```

Comment:
The class name logic for the 'Sort alphabetically' button is slightly redundant. The class 'button' is repeated twice. It can be simplified to `className={`button ${sortMethod === SORT_BY_ALPHABETICAL ? 'is-info' : 'is-info is-light'}`}`.